exporter/elasticsearchexporter/data_stream_router.go

```diff
@@ -6,12 +6,45 @@ package elasticsearchexporter // import "github.com/open-telemetry/opentelemetry
 import (
 	"fmt"
 	"regexp"
+	"strings"
+	"unicode"
 
 	"go.opentelemetry.io/collector/pdata/pcommon"
 )
 
 var receiverRegex = regexp.MustCompile(`/receiver/(\w*receiver)`)
 
+const (
+	maxDataStreamBytes       = 100
+	disallowedNamespaceRunes = "\\/*?\"<>| ,#:"
+	disallowedDatasetRunes   = "-\\/*?\"<>| ,#:"
+)
+
+// Sanitize the datastream fields (dataset, namespace) to apply restrictions
+// as outlined in https://www.elastic.co/guide/en/ecs/current/ecs-data_stream.html
+func sanitizeDataStreamField(field, disallowed string, otel bool) string {
+	// For Dataset, the naming convention for datastream is expected to be "logs-[dataset].otel-[namespace]".
+	// This is in order to match the built-in logs-*.otel-* index template.
+	var suffix string
+	if otel {
+		suffix += ".otel"
+	}
```

Comment:
move this back out of sanitization func, so that this func doesn't need to know about otel.